src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5569,16 +5630,64 @@ void markMembersAsNotNull(int receiverSlot, TypeSymbol type, string memberName,
             {
                 var (parameter, _, parameterAnnotations, _) = GetCorrespondingParameter(i, parametersOpt, argsToParamsOpt, expanded);
 
+                BoundExpression argument = arguments[i];
+                InterpolatedStringHandlerData? possibleInterpolationData = argument switch
+                {
+                    BoundInterpolatedString { InterpolationData: { } data } => data,
+                    BoundBinaryOperator { InterpolatedStringHandlerData: { } data } => data,
+                    _ => null
+                };
+
+                if (possibleInterpolationData is ({ ArgumentPlaceholders.Length: > 1 }
+                                                  or { ArgumentPlaceholders.Length: 1, HasTrailingHandlerValidityParameter: false })
+                                                 and var interpolationData)
+                {
+                    foreach (var placeholder in interpolationData.ArgumentPlaceholders)
+                    {
+                        switch (placeholder.ArgumentIndex)
+                        {
+                            case BoundInterpolatedStringArgumentPlaceholder.TrailingConstructorValidityParameter:
+                            case BoundInterpolatedStringArgumentPlaceholder.UnspecifiedParameter:
+                                break;
+                            case BoundInterpolatedStringArgumentPlaceholder.InstanceParameter:
+                                int slot = getInterpolatedStringHandlerPlaceholderSlot(placeholder, receiver);
+                                State[slot] = receiverState;
```

Comment:
>State[slot] = receiverState; [](http://example.com/codeflow?start=32&length=28)

It is not clear to me why overwriting the state for the real receiver's slot is the right thing to do.